runtime/near-wallet-contract/src/lib.rs

```diff
@@ -0,0 +1,62 @@
+#![doc = include_str!("../README.md")]
+use near_vm_runner::ContractCode;
+use std::sync::{Arc, OnceLock};
+
+/// Temporary (placeholder) Wallet Contract.
+pub fn wallet_contract() -> Arc<ContractCode> {
+    static CONTRACT: OnceLock<Arc<ContractCode>> = OnceLock::new();
+    CONTRACT.get_or_init(|| Arc::new(read_contract())).clone()
+}
+
+/// Include the WASM file content directly in the binary at compile time.
+fn read_contract() -> ContractCode {
+    #[cfg(feature = "nightly")]
+    let code = include_bytes!("../res/wallet_contract.wasm");
+
+    #[cfg(not(feature = "nightly"))]
+    let code = &[];
+
+    ContractCode::new(code.to_vec(), None)
+}
+
+/// near[wallet contract hash]
+pub fn wallet_contract_magic_bytes() -> Arc<ContractCode> {
+    static CONTRACT: OnceLock<Arc<ContractCode>> = OnceLock::new();
+    CONTRACT
+        .get_or_init(|| {
+            let wallet_contract_hash = *wallet_contract().hash();
+            let magic_bytes = format!("near{}", wallet_contract_hash.to_string());
+            Arc::new(ContractCode::new(magic_bytes.into(), None))
+        })
+        .clone()
+}
+
+#[cfg(feature = "nightly")]
+#[cfg(test)]
+mod tests {
+    use crate::{wallet_contract, wallet_contract_magic_bytes};
+    use near_primitives_core::hash::CryptoHash;
+    use std::str::FromStr;
+
+    const WALLET_CONTRACT_HASH: &'static str = "5wJJ2YaCq75kVSfx8zoZpevg1uLAn4h7nqUd2njKUEXe";
+    const MAGIC_BYTES_HASH: &'static str = "31PSU4diHE4cpWju91fb2zTqn5JSDRZ6xNGM2ub8Lgdg";
+
+    #[test]
+    fn check_wallet_contract() {
```

Comment:
Should these hash-checking tests be marked with `ignore` until the contract can be built with a reproducible hash?